hummingbot/hummingbot
# feat/order age from creation timestamp

**Before submitting this PR, please make sure**:


**A description of the changes proposed in the pull request**:
Change in the implementation of the order age calculation to stop using the timestamp that is part of the order_id and start using the new creation_timestamp field in the orders.


**Tests performed by the developer**:
Added unit tests.
Manually tested using a PMM strategy connected to paper trade Binance spot connector.


**Test Summary**:
- Create and start a PMM strategy using AscendEX, Binance, and Binance_paper_trade connector
- Run status --live and monitor the age of each order created: Confirmed that each order gets cancelled according to the specified order refresh time
- Filled event: Ok
- Run history --verbose to check the order_age
- Compare the timestamp saved in the CSV and sqlite file (filled table): Match
- Compare the order age of the filled order using the data on history vs order age in the csv file and using the creation timestamp and filled timestamp in the sqlite file: Order age matched
- Create an Avellaneda MM strategy, Liquidity Mining strategy, Perpetual MM, Aroon Osicillator and Hedge strategy then repeat the steps: Confirm that order age recorded in the history, CSV, and Sqlite file are matched
- Confirmed that all exchanges that support paper trade working as expected with order age.


**Tips for QA testing**:
- Check status command for the strategies report age correctly.
- Check age in trade events stored in the DB and in CSV files is correct.


This PR is created based on https://github.com/CoinAlpha/hummingbot/pull/61
